### PR TITLE
Modbus logging v6

### PR DESCRIPTION
**Pull request by glongo**
Updated of #2896 

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2096

Describe changes:
- fix a NULL pointer deference

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

- PR glongo: https://buildbot.openinfosecfoundation.org/builders/glongo/builds/149
- PR glongo-pcap: https://buildbot.openinfosecfoundation.org/builders/glongo-pcap/builds/13

### CODE REVIEW

**Comment by victorjulien** [comment]
@dbcfd could you have another look at this version?

**Comment by dbcfd** [comment]
Pcaps that were crashing when I was testing no longer crash on this branch.

**Comment by victorjulien** [comment]
Can you add to the PR:
1. alert record with modbus
2. modbus record with 'dict'
3. modbus record with 'array'

**Comment by victorjulien** [comment]
Add it directly to the PR please, no links to other places.

File: doc/userguide/output/eve/eve-json-output.rst [hunk 1]
```diff
@@ -110,6 +110,8 @@ The most common way to use this is through 'EVE', which is a firehose approach w
             #md5: [body, subject]
 
         - ssh
+        - modbus:
+            array-as-dict: yes       # log values as an integer indexed dict (ELK compliant)
```
Comment (victorjulien) [inline doc/userguide/output/eve/eve-json-output.rst]: if I read the code correctly, this controls the type of a field in the json output. I think having a single field that can have multiple types is not good practice.

Comment (victorjulien) [reply doc/userguide/output/eve/eve-json-output.rst]: Explanation of the reason for this option to exist is missing.

**Comment by daviddiallo** [comment]
PR is good for me, thanks @glongo.

**Pull request closed without merging**